src/systems/drawing-menu-system.js

```diff
@@ -0,0 +1,200 @@
+import { waitForDOMContentLoaded } from "../utils/async-utils";
+
+/**
+ * Drawing Menu System
+ * A tool that allows drawing on networked-drawing components.
+ * @namespace drawing
+ * @component pen
+ */
+
+function almostEquals(epsilon, u, v) {
+  return Math.abs(u.x - v.x) < epsilon && Math.abs(u.y - v.y) < epsilon && Math.abs(u.z - v.z) < epsilon;
+}
+
+export class DrawingMenuSystem {
+  constructor(sceneEl) {
+    this.sceneEl = sceneEl;
+
+    this.sceneEl.addEventListener("stateremoved", this.hideMenus.bind(this));
+
+    this.lastIntersection = new THREE.Vector3();
+
+    this.drawingMenus = [];
+    this.setDirty = this.setDirty.bind(this);
+    this.dirty = true;
+
+    // TODO: Use the MutationRecords passed into the callback function to determine added/removed nodes!
+    this.observer = new MutationObserver(this.setDirty);
```

Comment:
Do you want to be observing the whole scene? It seems like you only care when new `.drawing-menu`'s get added, so I wonder if they should have a component similar to `is-remote-hover-target` on it that adds itself to a collection in `init` and then removes itself in `remove`